pkg/apis/sparkoperator.k8s.io/v1alpha1/types.go

```diff
@@ -231,13 +238,16 @@ type ApplicationStateType string
 
 // Different states an application may have.
 const (
-	NewState              ApplicationStateType = "NEW"
+	NewState              ApplicationStateType = ""
 	SubmittedState        ApplicationStateType = "SUBMITTED"
 	RunningState          ApplicationStateType = "RUNNING"
 	CompletedState        ApplicationStateType = "COMPLETED"
 	FailedState           ApplicationStateType = "FAILED"
 	FailedSubmissionState ApplicationStateType = "SUBMISSION_FAILED"
-	UnknownState          ApplicationStateType = "UNKNOWN"
+	PendingRetryState     ApplicationStateType = "PENDING_RETRY"
```

Comment:
As we discussed, let's use `PENDING_RERUN` for reruns as a result of spec updates?